### PR TITLE
docker client & pause container find

### PR DESCRIPTION
**Pull request by John-Lin**
- finding pause container
- docker client operate functions

### CODE REVIEW

File: docker/pause-container.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	"regexp"
+
+	"docker.io/go-docker/api/types"
+)
+
+func FindK8SPauseContainerID(containers []types.Container, PodName, Namespace, PodUUID string) (string, error) {
+	for _, container := range containers {
+		// ex: k8s_POD_myinit_default_05ab36d8-65aa-11e8-b35e-42010af00248_0
+		pattern := fmt.Sprintf("k8s_POD_%s_%s_%s_\\d+", PodName, Namespace, PodUUID)
+		r, err := regexp.Compile(pattern)
```
Comment (hwchiu) [inline docker/pause-container.go]: Move out the regexp to out of the for loop, we don't need to compile it every time.

File: docker/pause-container.go [hunk 2]
```diff
@@ -8,10 +8,10 @@ import (
...
 
 func FindK8SPauseContainerID(containers []types.Container, PodName, Namespace, PodUUID string) (string, error) {
+	// ex: k8s_POD_myinit_default_05ab36d8-65aa-11e8-b35e-42010af00248_0
+	pattern := fmt.Sprintf("k8s_POD_%s_%s_%s_\\d+", PodName, Namespace, PodUUID)
+	r, err := regexp.Compile(pattern)
 	for _, container := range containers {
-		// ex: k8s_POD_myinit_default_05ab36d8-65aa-11e8-b35e-42010af00248_0
-		pattern := fmt.Sprintf("k8s_POD_%s_%s_%s_\\d+", PodName, Namespace, PodUUID)
-		r, err := regexp.Compile(pattern)
 		if err != nil {
```
Comment (hwchiu) [inline docker/pause-container.go]: The err checking should also be moved out.

**Comment by John-Lin** [comment]
docker client is difficult to write tests.

**Comment by John-Lin** [comment]
init container can get pod name, namespace, pod uuid from downward API. we could use these to send to network controller server. the network controller using these information to find pause container ID and network namespace path.
example pod yaml
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: myinit
spec:
  containers:
  - name: app-container
    image: busybox
    command:
      - env
  initContainers:
  - name: init-container
    image: busybox
    command:
    - env
    env:
    - name: MY_POD_NAME
      valueFrom:
        fieldRef:
          fieldPath: metadata.name
    - name: MY_POD_NAMESPACE
      valueFrom:
        fieldRef:
          fieldPath: metadata.namespace
    - name: MY_POD_UUID
      valueFrom:
        fieldRef:
          fieldPath: metadata.uid
    - name: MY_POD_IP
      valueFrom:
        fieldRef:
          fieldPath: status.podIP
```

**Comment by hwchiu** [comment]
Move out the err check function of regexp from `for loop` and add some invalid testing.

**Pull request merged**